### PR TITLE
#Cu-20465559 #xcvm-276

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
for https://app.clickup.com/t/20465559/XCVM-276
based on https://github.com/ComposableFi/composable/pull/2971

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [XCVM-276 Write simple test to ensure IBC hooks work](https://app.clickup.com/t/20465559/XCVM-276)

File: code/parachain/frame/cosmwasm/src/mapping.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			version.to_string(),
+			connection_hops
+				.get(0)
+				.expect("by spec there is at least one connection; qed")
+				.to_string(),
+		),
+	})
+}
+
+pub fn map_order(order: IbcOrder) -> Result<CwOrder, IbcError> {
```
Comment (benluelo) [inline code/parachain/frame/cosmwasm/src/mapping.rs:39]: This fn always returns Ok, is that intended?

File: code/parachain/frame/cosmwasm/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #![recursion_limit = "256"]
 #![feature(sync_unsafe_cell)]
+#![feature(trait_alias)]
```
Comment (benluelo) [inline code/parachain/frame/cosmwasm/src/lib.rs:3]: Where is this used?

File: code/parachain/frame/cosmwasm/src/mapping.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	cosmwasm_vm::cosmwasm_std::IbcTimeoutBlock {
+		revision: revision_number,
+		height: revision_height,
+	}
+}
+
+pub fn to_cosmwasm_timestamp(
+	timestamp: ibc::timestamp::Timestamp,
+) -> cosmwasm_vm::cosmwasm_std::Timestamp {
+	cosmwasm_vm::cosmwasm_std::Timestamp::from_nanos(timestamp.nanoseconds())
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply code/parachain/frame/cosmwasm/src/mapping.rs:58]: https://composablefinance.slack.com/archives/C03B8BN8ULX/p1675082478415239 see comments.

File: code/parachain/frame/cosmwasm/src/mapping.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/parachain/frame/cosmwasm/src/mapping.rs:39]: will remove. before that Centauri had 3 values and Coswasm has 2. Now both has 2.

Comment (dzmitry-lahoda) [reply code/parachain/frame/cosmwasm/src/mapping.rs:39]: fixing

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by benluelo — APPROVED** [review]

**Review by aeryz — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Review by cor — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Review by obsessed-cake — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@KaiserKarel what is needed to merge this?

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Comment by dzmitry-lahoda** [comment]
Retried several times https://github.com/ComposableFi/composable/actions/runs/4052211142/jobs/6971607209 . I see Nix HTTP 200 fail.

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Review by obsessed-cake — APPROVED** [review]

**Review by benluelo — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Pull request merged**